### PR TITLE
Lifei/v1.0 gooose configure cli

### PR DESCRIPTION
**Pull request by lifeizhou-ap**
**What**
Implemented `goose --version` and `goose configure`

**goose configure**
Usage
- goose configure
- goose configure [profile name]

Rules
-  If the profile name is not passed in the command, it will ask the user to type in the profile name.
- If the profile has not been set before, it will create a new profile in the profile file
- If the profile has been set before, the user will be prompt with the existing value as default and be asked to enter the new value and the modified profile will be saved to the profile file
- After the end of the configure, it will use the profile to call provider and show user a welcome message.

Input values:
- provider name (if the command does not contain the profile name)
- provider (user chooses from a list)
- processor
- accelerator 
- provider specific configuration
   - open ai: api key if it is not specified in the environment variables
   - databricks: host and token if they are not specified in the environment variables

Screenshots
- Add profile
add_profile

- Edit profile
edit_profile_image

Note:
- I have not added the max_token and temperature as the input yet. We can add it in if they are required. 
- currently I used `~/.config/goose/profile-1.0.yaml file so that it does not conflict with the existing goose usage

### CODE REVIEW

File: crates/goose-cli/src/commands/configure.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    pub accelerator: Option<String>,
+}
+
+pub fn handle_configure(options: ConfigOptions) -> Result<(), Box<dyn Error>> {
+    ctrlc::set_handler(move || {}).expect("setting Ctrl-C handler");
+    
+    cliclack::clear_screen()?;
+
+    cliclack::intro(style(" create-app ").on_cyan().black())?;
+    let provider = prompt(options.provider, "Enter provider name:");
```
Comment (baxen) [inline crates/goose-cli/src/commands/configure.rs]: we should make use of all the cliclack primitives here i think!
see e.g. https://github.com/fadeevab/cliclack/blob/main/examples/basic.rs#L21
or also the code gen version from goose:
https://github.com/block/goose/pull/241/files

File: crates/goose-cli/src/commands/configure.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    pub host: Option<String>,
+    pub token: Option<String>,
+    pub processor: Option<String>,
+    pub accelerator: Option<String>,
+}
+
+pub fn handle_configure(options: ConfigOptions) -> Result<(), Box<dyn Error>> {
+    ctrlc::set_handler(move || {}).expect("setting Ctrl-C handler");
+    
+    cliclack::clear_screen()?;
```
Comment (baxen) [inline crates/goose-cli/src/commands/configure.rs]: nit: i dont think we should clear the screen for this

File: crates/goose-cli/src/commands/configure.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    let accelerator = prompt(options.accelerator, "Enter accelerator:");
+
+    let final_config = ConfigOptions {
+        provider: Some(provider),
+        host: Some(host),
+        token: Some(token),
+        processor: Some(processor),
+        accelerator: Some(accelerator),
+    };
+    match save_to_yaml(&final_config) {
```
Comment (baxen) [inline crates/goose-cli/src/commands/configure.rs]: imo would be good to test this here by calling provider.complete() to confirm it works and share a fun message

File: crates/goose-cli/src/commands/configure.rs (hunk 1 above)
Comment (lifeizhou-ap) [reply crates/goose-cli/src/commands/configure.rs]: this `prompt` function is defined by ourself that wrapped the cliclack `input` https://github.com/block/goose/blob/af7ce7c82350f5dd6ad3e7fcbe9a6e468704a325/crates/goose-cli/src/commands/configure.rs#L46

**Review by baxen — APPROVED** [review]

**Pull request merged**